### DISCUSSION TITLE
__setInitialJobParameters method in JobDB seems very weird

### DISCUSSION THREAD

**Discussion by simon-mazenoux** [DIRACGrid/DIRAC#6983] [Q&A] [answered]
When tinkering around the submission process, I noticed a method which looks very weird to me.

```python
def __setInitialJobParameters(self, classadJob, jobID):
        """Set initial job parameters as was defined in the Classad"""

        # Extract initial job parameters
        parameters = {}
        if classadJob.lookupAttribute("Parameters"):
            parameters = classadJob.getDictionaryFromSubJDL("Parameters")
        return self.setJobParameters(jobID, list(parameters.items()))
```

This method calls the setJobParameters where the JDL contains the attribute Parameters. However, this argument is supposed to be removed from the JDL when parametric jobs are unserialized in the JobManagerHandler.

Moreover, the method setJobParameters is also used in the JobStateUpdateHandler in the export_setJobParameters, which is itself use by the JobReport class.

I'm guessing that the __setInitialJobParameters was a result of a misunderstanding between parameters from parametric jobs and and jobParameters.

Do you think I'm right, or is there something that I'm missing

**Comment by atsareg** [comment, marked as answer]
What is meant in the __setInitialJobParameters and in general what we call the Job Parameters have nothing to do with the parametric jobs. It is true that there is an unfortunate double use of the term "Parameters" in both cases. I think for the moment it should be just clarified for the developers and not really recode it to remove this ambiguity.
